include/lsst/afw/detection/Footprint.h

```diff
@@ -305,32 +206,26 @@ class Footprint : public lsst::daf::base::Citizen,
      *                       NOT the same as the footprint's bounding box.
      *  @param[in]  doClip   If true, clip the new footprint to the region bbox before returning it.
      */
-    PTR(Footprint) transform(
+    std::shared_ptr<Footprint> transform(
```

Comment:
I wonder if we should return `unique_ptr` here instead of `shared_ptr`.

Argument in favor:
 - `shared_ptr`s can be constructed from `unique_ptr`s, but not vice versa, so `unique_ptr` gives the caller more flexibility.

Argument against:
 - If we return `shared_ptr`, we can use `make_shared` to construct a `shared_ptr` more efficiently than we could if we went through `unique_ptr`, and we'll probably almost always put our `Footprint`s in `shared_ptr`s, since that's how `SourceRecord` holds them.
 - I'm not sure Swig understands `unique_ptr`.

Given that constructing a `Footprint` (let alone transforming one) is much more expensive than the gain we get from using `make_shared` and we're already considering doing this directly in pybind11, I think `unique_ptr` is clearly better.  That holds for any method that you currently have returning a `shared_ptr<Footprint>`.